p2panda-rs/src/document/document_view_id.rs

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use crate::hash::{Hash, HashError};
+use crate::Validate;
+
+/// The identifier of a document view.
+///
+/// Contains the hashes of the document graph tips which is all the information we need to reliably
+/// recreate the document at this certain point in time.
+///
+/// ```text
+/// The document with the following operation graph has the id "2fa.." and six different document
+/// view ids, meaning that this document can be represented in six versions:
+///
+/// 1. ["2fa"]
+/// 2. ["de8"]
+/// 3. ["89c"]
+/// 4. ["eff"]
+/// 5. ["de8", "eff"]
+/// 6. ["89c", "eff"]
+///
+/// [CREATE] (Hash: "2fa..") <-- [UPDATE] (Hash: "de8..") <-- [UPDATE] (Hash: "89c..")
+///                         \
+///                          \__ [UPDATE] (Hash: "eff..")
+/// ```
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct DocumentViewId(Vec<Hash>);
+
+impl DocumentViewId {
+    /// Create a new document view id.
+    pub fn new(graph_tips: Vec<Hash>) -> Self {
+        Self(graph_tips)
+    }
+
+    /// Get the graph tip hashes of this view id.
+    pub fn graph_tips(&self) -> &[Hash] {
+        self.0.as_slice()
+    }
+}
+
+impl Validate for DocumentViewId {
+    type Error = HashError;
+
+    fn validate(&self) -> Result<(), Self::Error> {
+        for hash in &self.0 {
+            hash.validate()?;
+        }
+
+        Ok(())
+    }
+}
+
+/// Convenience method converting a single hash into a document view id.
+///
+/// Converts a `Hash` instance into a `DocumentViewId`, assuming that this document view only
+/// consists of one graph tip hash.
+impl From<Hash> for DocumentViewId {
+    fn from(hash: Hash) -> Self {
+        Self::new(vec![hash])
+    }
+}
+
+/// Convenience method converting a hash string into a document view id.
+///
+/// Converts a hash string into a `DocumentViewId`, assuming that this document view only consists
+/// of one graph tip hash.
+impl FromStr for DocumentViewId {
```

Comment:
You could even remove this and just do 

```rs
let document_id: DocumentViewId = "0020d3235c8fe6f58608200851b83cd8482808eb81e4c6b4b17805bba57da9f16e79".parse::<Hash>().unwrap().into();
```

Rust! `0_0`

Reply:
I was wondering if this is possible! Thanks for showing!